pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -43,6 +43,55 @@ func TestParseLocation(t *testing.T) {
 	assert.Equal(t, "example.com:5000/with/path", location)
 }
 
+func TestParseSubDomainPrefix(t *testing.T) {
+	var err error
+	var subdomainprefix string
+
+	// invalid subdomainprefix
+	_, err = parseSubDomainPrefix("https://example.com")
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "invalid subdomainprefix 'https://example.com': does not begin with '*.'")
+
+	_, err = parseSubDomainPrefix("john.doe@example.com")
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "invalid subdomainprefix 'john.doe@example.com': does not begin with '*.'")
+
+	/*
+			subdomainprefix, err = parseSubDomainPrefix("example.com")
+			assert.NotNil(t, err)
+			assert.Contains(t, err.Error(), "invalid subdomainprefix 'example.com': does not begin with '*.'")
+
+			subdomainprefix, err = parseSubDomainPrefix("example.com/")
+			assert.NotNil(t, err)
+			assert.Contains(t, err.Error(), "invalid subdomainprefix 'example.com/': does not begin with '*.'")
+
+			subdomainprefix, err = parseSubDomainPrefix("example.com//////")
+			assert.NotNil(t, err)
+			assert.Contains(t, err.Error(), "invalid subdomainprefix 'example.com//////': does not begin with '*.'")
+
+		subdomainprefix, err = parseSubDomainPrefix("example.com:5000/with/path")
+		assert.NotNil(t, err)
+		assert.Contains(t, err.Error(), "invalid subdomainprefix 'example.com:5000/with/path': does not begin with '*.'")
+	*/
```

Comment:
I'm getting `ineffassign` errors for these commented tests in `make validate`. Not quite sure why that is yet.